lib/iris/cube.py

```diff
@@ -1919,6 +1957,73 @@ def coord_system(self, spec=None):
 
         return result
 
+    def _a_meshcoord(self):
+        mesh_coords = self.coords(mesh_coords=True)
+        if mesh_coords:
+            result = mesh_coords[0]
+        else:
+            result = None
+        return result
+
+    def mesh(self):
```

Comment:
Well, my view on this was that it isn't a simple pragmatic choice whether to use `cube.mesh` or `cube.mesh()` as they "should" indicate different kinds of things.
I.E. in my head, an instance property is a simple piece of the object state, which you can expect to be a simple get/set., whereas methods are used for derived stuff and calculations.

So in this case, a MeshCoord has mesh + location + axis properties, because they are the core parts of its state/identity, whereas a Cube should have mesh() location() etc because they are calculated values, not fundamental to the object itself.

Obviously, all that isn't a 'real' distinction in Python, unlike other O-O languages.
Do you have any sympathy with this idea ?

Reply:
@pp-mo Regardless, what your effectively saying is that you're happy for force users to do this?
```python
cube.location() == coord.location and cube.mesh() == coord.mesh
``` 
Which makes me fundamentally sad to my core; sorry but it's true :disappointed: :-1: 

Whereas, what I'm suggesting is this...
```python
cube.location == coord.location and cube.mesh == coord.mesh
```
Yes, please :smile: :+1:

Reply:
Well I guess so, if these things are not the same _kind_ of thing for the cube and the coord.
Which is why we have cube.coords() and not cube,coords.   Like dict.values().
But I can see I just won't win this argument.